packages/userstorage/src/browser/user-storage-service-filesystem.ts

```diff
@@ -52,8 +52,10 @@ export class UserStorageServiceFilesystemImpl implements UserStorageService {
         this.userStorageFolder.then(folder => {
             if (folder) {
                 for (const change of fileChanges) {
-                    const userStorageUri = UserStorageServiceFilesystemImpl.toUserStorageUri(folder, change.uri);
-                    uris.push(userStorageUri);
+                    if (change.uri.toString().startsWith(folder.toString())) {
```

Comment:
I think you should also make sure that the `onUserStorageChangedEmitter` isn't fired if no fileChanges were related to the user storage folder, thus having en empty `uris` array.

Reply:
Done, verifying to trigger the fire only when there is some uris